packages/copy-widget-code/src/modules/widgetCode/sagas.js

```diff
@@ -0,0 +1,71 @@
+import {all, call, put, select, takeEvery, takeLatest} from 'redux-saga/effects'
+import {rest, notification} from 'tocco-app-extensions'
+import {js, consoleLogger} from 'tocco-util'
+
+import {generateWidgetCode} from '../../utils/widgetCode'
+import * as actions from './actions'
+
+export const inputSelector = state => state.input
+export const widgetCodeSelector = state => state.widgetCode
+
+export default function* sagas() {
+  yield all([
+    takeEvery(actions.FETCH_WIDGET_CONFIG, fetchWidgetConfig),
+    takeLatest(actions.COPY_WIDGET_CODE, copyWidgetCode)
+  ])
+}
+
+export const getSingleWidgetConfigKey = selection => {
```

Comment:
should we move this function to the `selection` app-extension? you copied it and I need the exact same function in the widget config edit action and many other actions might need it as well. The only thing that's variable is the expected entity name.